### PR TITLE
YAML based config

### PR DESCRIPTION
**Pull request by murrant**
YAML based initial configuration files.
https://github.com/librenms/librenms/pull/13259

Sets in the database so they are persistent through-out the install and can be configured in the Web UI or lnms config:set
.yaml files will only be applied to a clean database.

### CODE REVIEW

**Comment by murrant** [comment]
Anything blocking this from being merged?

File: doc/notes/edit-config.md [hunk 1]
```diff
@@ -1,11 +1,36 @@
...
+
+You can edit the running configuration via the LibreNMS web UI or `lnms config:set`
+
+```bash
+docker-compose exec librenms lnms config:set page_refresh 300
+```
+
+## Re-apply yaml config settings during deploy
+
+Set REAPPLY_YAML_CONFIG=1 to overwrite any settings that are set during initial config
```
Comment (crazy-max) [inline doc/notes/edit-config.md:23]:
```suggestion
Set `REAPPLY_YAML_CONFIG=1` to overwrite any settings that are set during initial config
```

File: doc/notes/syslog-ng.md [hunk 2]
```diff
@@ -11,9 +11,8 @@ docker run -d --name librenms_syslog \
...
 ```
 
-You have to create a configuration file to enable syslog in LibreNMS too. Create a file called for example `/data/config/syslog.php` with this content :
+You have to create a configuration file to enable syslog in LibreNMS too. Create a file called for example `/data/config/syslog.yaml` with this content :
 
-```php
-<?php
-$config['enable_syslog'] = 1;
+```yaml
+enable_syslog: true;
```
Comment (crazy-max) [inline doc/notes/syslog-ng.md:17]:
```suggestion
enable_syslog: true
```

File: rootfs/etc/cont-init.d/03-config.sh [hunk 3]
```diff
@@ -113,78 +113,60 @@ DB_PASSWORD=${DB_PASSWORD}
...
 
-# Config : Memcached
-if [ -n "${MEMCACHED_HOST}" ]; then
-    cat > ${LIBRENMS_PATH}/config.d/memcached.php <<EOL
-<?php
-\$config['memcached']['enable'] = true;
-\$config['memcached']['host'] = '${MEMCACHED_HOST}';
-\$config['memcached']['port'] = ${MEMCACHED_PORT};
-EOL
-fi
```
Comment (crazy-max) [inline rootfs/etc/cont-init.d/03-config.sh:172]: Any reason to remove the memcached config?

Comment (murrant) [reply rootfs/etc/cont-init.d/03-config.sh:172]: These settings don't exist in LibreNMS.

File: README.md [hunk 4]
```diff
@@ -352,11 +378,10 @@ $ docker run -d --name librenms_syslog \
...
 
 You have to create a configuration file to enable syslog in LibreNMS too. Create
-a file called for example `/data/config/syslog.php` with this content:
+a file called for example `/data/config/syslog.yaml` with this content :
 
-```php
-<?php
-$config['enable_syslog'] = 1;
+```yaml
+enable_syslog: true;
```
Comment (crazy-max) [inline README.md:384]:
```suggestion
enable_syslog: true
```

File: README.md [hunk 5]
```diff
@@ -260,22 +260,48 @@ $ docker-compose pull
...
+You can edit the running configuration via the LibreNMS web UI or `lnms config:set`
 
-### Edit configuration
+```bash
+docker-compose exec librenms lnms config:set page_refresh 300
+```
+
+### Re-apply yaml config settings during deploy
+
+Set REAPPLY_YAML_CONFIG=1 to overwrite any settings that are set during initial config
```
Comment (crazy-max) [inline README.md:288]:
```suggestion
Set `REAPPLY_YAML_CONFIG=1` to overwrite any settings that are set during initial config
```

**Review by crazy-max — CHANGES_REQUESTED** [review]
Can you also update the TOC in the README? Thanks

File: README.md (hunk 4 above)
Comment (murrant) [reply README.md:384]: That sneaky little bugger.

**Review by crazy-max — APPROVED** [review]
LGTM thanks!

**Pull request merged**